packages/core/src/shared/clients/docdbClient.ts

```diff
@@ -214,20 +214,49 @@ export class DefaultDocumentDBClient {
         return !!response.DBInstance
     }
 
-    public async listResourceTags(arn: string): Promise<DocDB.Tag[]> {
-        const command = new DocDB.ListTagsForResourceCommand({ ResourceName: arn })
-        const response = await this.executeCommand<DocDB.ListTagsForResourceCommandOutput>(command)
-        return response.TagList ?? []
+    public async listResourceTags(arn: string): Promise<Record<string, string | undefined> | undefined> {
+        if (isElasticCluster(arn)) {
+            const command = new DocDBElastic.ListTagsForResourceCommand({ resourceArn: arn })
+            const response = await this.executeElasticCommand<DocDBElastic.ListTagsForResourceCommandOutput>(command)
+            return response.tags
+        } else {
+            const command = new DocDB.ListTagsForResourceCommand({ ResourceName: arn })
+            const response = await this.executeCommand<DocDB.ListTagsForResourceCommandOutput>(command)
+            const tagMap = response.TagList?.reduce(
+                (map, tag) => {
+                    map[tag.Key!] = tag.Value
```

Comment:
Rather than force unwrapping the `tag.Key` you can simply check for null and be safer.

```js
if (tag.Key) {
  map[tag.Key] = tag.Value
}
return map
```

Reply:
I don't know why they defined Key as optional, as it's a required value when adding a tag, so it should never be undefined.